mbuild/tests/test_compound.py

```diff
@@ -1088,6 +1088,18 @@ def test_bond_graph(self, ch3):
             for particle in ch3_nobonds.particles()
         )
 
+    def test_is_independent(self):
+        from mbuild.lib.molecules import WaterSPC
```

Comment:
@daico007 , this looks great! One test that might make it easy for others to understand the behavior of this would be to have a system where you can show adding a bond between two previously independent compounds and that it now invalidates the is_independent check. 

LGTM!

Reply:
sounds cool! I will add that test in.

Reply:
Extra test added!